### PR TITLE
allow subtracting from aggregate for when participation is high

### PR DESCRIPTION
**Pull request by etan-status**
When the aggregate public key of full participation is known, and participation is high, it is worthwhile to subtract individual keys from the known full participation aggregate public key.

Add a corresponding `fastAggregateVerify` overload to support that.

### CODE REVIEW

File: blscurve/blst/blst_min_pubkey_sig_core.nim [hunk 1]
```diff
@@ -187,6 +187,17 @@ template genAggregatorProcedures(
...
     return true
 
+  proc subtractAll*(dst: var BaseType, elems: openArray[BaseType]) =
+    ## Subtracts all ``elems[0..<elems.len`` from ``dst``.
+    if len(elems) == 0:
+      return
+    var agg{.noinit.}: Aggregate
+    agg.init(elems[0])
+    agg.aggregate(elems.toOpenArray(1, elems.high))
+    agg.point.`blst _ p1_or_p2 _ cneg`(cbit = 1)
```
Comment (etan-status) [inline blscurve/blst/blst_min_pubkey_sig_core.nim:197]: not sure what `cbit` does. in the bindings, there are overloads without a `cbit` that redirect to `cbit = 1`

File: blscurve/blst/blst_min_pubkey_sig_core.nim [hunk 2]
```diff
@@ -187,6 +187,17 @@ template genAggregatorProcedures(
     dst.finish(agg)
     return true
 
+  proc subtractAll*(dst: var BaseType, elems: openArray[BaseType]) =
+    ## Subtracts all ``elems[0..<elems.len`` from ``dst``.
```
Comment (etan-status) [inline blscurve/blst/blst_min_pubkey_sig_core.nim:191]:
```suggestion
    ## Subtracts all ``elems[0..<elems.len]`` from ``dst``.
```

**Review by mratsim — APPROVED** [review]

**Comment by etan-status** [comment]
Thanks for the review, @mratsim :)

File: blscurve/bls_sig_min_pubkey.nim [hunk 3]
```diff
@@ -256,3 +256,18 @@ func fastAggregateVerify*[T: byte|char](
   if not aggAffine.aggregateAll(publicKeys):
     return false
   return coreVerifyNoGroupCheck(aggAffine, message, signature, DST)
+
+func fastAggregateVerify*[T: byte|char](
```
Comment (arnetheduck) [inline blscurve/bls_sig_min_pubkey.nim:260]: I think in general, we should be getting rid of these pre-composed helpers that add surface without adding significant functionality..

**Pull request merged**